libclc/CMakeLists.txt

```diff
@@ -370,7 +385,10 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
     elseif( ARCH STREQUAL amdgcn )
       set( opt_flags -O3 --amdgpu-oclc-reflect-enable=false )
     elseif( ARCH STREQUAL x86_64)
```

Comment:
This is the second case I see where we're still doing separate checks for `x86_64` and `aarch64` which share a lot of logic. I think we might want to think about having a common variable set for when we're compiling for native-cpu and check that at the top level.

Then, if required, we could do additional target-specific logic inside that.

``` cmake
elseif( NATIVE_CPU )
  set( opt_flags -O3 )
  set( has_distinct_generic_addrspace FALSE )
```

``` cmake
elseif( NATIVE_CPU )
  list( APPEND flags -Xclang -fsycl-is-native-cpu)
  if( ARCH STREQUAL x86_64)
    list( APPEND flags -Xclang -target-feature -Xclang +avx -Xclang -target-feature -Xclang +avx512f)
  endif()
```

... that sort of thing.

That would keep the number of changes if/when we add a third/fourth native-cpu target to a minimum.